lib/action_client/base.rb

```diff
@@ -87,6 +93,10 @@ def build_request(method:, path: nil, url: nil, query: {}, headers: {}, **option
         request.headers[key] = value
       end
 
+      if block
+        @middleware.unshift ActionClient::Callbacks::AfterSubmit, &block
```

Comment:
@georgebrock I've totally re-worked this branch to reduce the number of concepts at play.

There is no more a separate Array of after submit actions. Everything operates as a middleware.

I think it's much more simple, and could use a fresh review.

Reply:
:+1: I've marked the existing comment threads as resolved, and will look through from the start.